PCF-ADR-0003-codeflare-release-process.md

```diff
@@ -0,0 +1,117 @@
+| Supersedes     | N/A                      |
+| Superseded by: | N/A                      |
+| Issues         |                          |
+| Other docs:    | PCF-ADR-002              |
+
+## What
+
+This ADR introduces a unified release process, testing, and versioning scheme for the CodeFlare project.
+
+## Why
+
+The CodeFlare project currently has different release processes and versioning schemes for all of it's subprojects. This creates confusion for users when they wish to use the stack -- what components versions should they be using? How do external dependencies like KubeRay factor in?
+There are also no standards around testing for the project
+This ADR aims to document the overall release process for Project CodeFlare.
+
+## Goals
+
+* Establish a common versioning scheme for all CodeFlare components
+* Establish a release process for the CodeFlare project
+* Establish guidelines for CodeFlare support matrix
+
+## Non-Goals
+
+* Require all components to restart versioning from scratch
+* Outline technical details for tests or CI infrastructure.
+
+## How
+
+We are proposing a unified release cycle for the CodeFlare project.
+We will release a new version of the CodeFlare stack after every sprint (currently every 3 weeks), following normal semantic versioning. All components under the CodeFlare umbrella are expected to start using semantic versioning for future releases. Components are not expected to start versioning again from scratch (i.e v0.0.1). As part of a new release for the CodeFlare Stack, we will update the support matrix which outlines specific component versions and versions for important dependencies such as KubeRay. The components to be documented are as follows:
+
+* CodeFlare Notebook Image
+* CodeFlare Operator
+  * The CodeFlare Operator version will be aligned with (i.e equivalent to) the CodeFlare stack version
+* CodeFlare SDK
+  * Supported TorchX version will be documented under the SDK.
+* InstaScale
+* MCAD
+* KubeRay
+  * On a new KubeRay release, CodeFlare will update it's supported version by the end of the next CodeFlare sprint
+
+### Images
+
+For published images, we are proposing repositories include the following three tags.
+Stable – This will be updated once every sprint as part of an official release
+Fast – This will be updated every time a new PR is merged into the main branch of a repository
+Normal semantic version tags – These will be created whenever components require a new image to be published. This may happen more than once per sprint.
+
+### Testing
+
+Each component will be required to implement unit and e2e tests that are run on all PRs.
+The CodeFlare operator will also include an integration test suite that will ensure stability of the overall stack.
+In addition to running the integration test suite on every PR, we will run it nightly in the CodeFlare operator repository with the `fast` tag for all underlying components
```

Comment:
We will only run these on every PR to the CodeFlare operator, right? I _think_ that's the case, and if so I think we could reword this sentence to make that more explicit.